botocore/credentials.py

```diff
@@ -550,19 +554,51 @@ def load(self):
         """
         if self._mapping['access_key'] in self.environ:
             logger.info('Found credentials in environment variables.')
-            access_key, secret_key = self._extract_creds_from_mapping(
-                self.environ, self._mapping['access_key'],
-                self._mapping['secret_key'])
-            token = self._get_session_token()
-            return Credentials(access_key, secret_key, token,
-                               method=self.METHOD)
+            fetcher = self._create_credentials_fetcher()
+            credentials = fetcher()
+            expiry_time = credentials['expiry_time']
+            if expiry_time is not None:
```

Comment:
So for the very first time you load credentials and you do not have an expiry time set, it will not ever try to refresh if it you add the expiry time variable later on? I'm just curious about the behavior. Have not really thought out what should be desired.